core/client/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -160,17 +156,10 @@ public static BlockInStream createUfsBlockInStream(FileSystemContext context, St
           closer.register(blockWorkerClient.lockUfsBlock(blockId, lockBlockOptions)).getResult();
       PacketInStream inStream;
       if (lockBlockResult.getLockBlockStatus().blockInAlluxio()) {
-        boolean local = blockWorkerClient.getDataServerAddress().getHostName()
-            .equals(NetworkAddressUtils.getClientHostName());
-        if (local && Configuration.getBoolean(PropertyKey.USER_SHORT_CIRCUIT_ENABLED)) {
```

Comment:
This was an optimization. This happens when the client started with reading from UFS (lockUfsBlock) but the block worker tells the client that the block is in Alluxio. Then the client tries to use short circuit read first.